mono/metadata/object.c

```diff
@@ -5294,9 +5316,12 @@ mono_value_box (MonoDomain *domain, MonoClass *klass, gpointer value)
 	vtable = mono_class_vtable (domain, klass);
 	if (!vtable)
 		return NULL;
```

Comment:
We must introduce MonoError to `mono_class_vtable` first, as we have absolutely no idea what the error was, that led to `mono_class_vtable` returning NULL. We can obviously return a dummy error, but that's not going to help. We can otherwise `g_assert (vtable)`, but as it's used way more broadly than `mono_object_new_pinned`, we cannot guarantee it's going to always make sense. Then we have exactly the same issue with `mono_nullable_box` 4 lines above, where we have no guarantee it might not return NULL, and this, without setting the error.

So IMHO the best solution here is to at least leave a comment `/* FIXME don't return NULL without setting error */`, and we come back on them as we migrate functions along the way, just as the other FIXME.